pkg/clusteragent/externalmetrics/autoscaler_watcher.go

```diff
@@ -76,19 +90,30 @@ func NewAutoscalerWatcher(refreshPeriod int64, autogenEnabled bool, autogenExpir
 	}
 
 	// Setup HPA
-	var autoscalerLister autoscaler_lister.HorizontalPodAutoscalerLister
+	var autoscalerLister cache.GenericLister
 	var autoscalerListerSynced cache.InformerSynced
 	if informer != nil {
-		autoscalerLister = informer.Autoscaling().V2beta1().HorizontalPodAutoscalers().Lister()
-		autoscalerListerSynced = informer.Autoscaling().V2beta1().HorizontalPodAutoscalers().Informer().HasSynced
+		hpaGVR, err := autoscalers.DiscoverHPAGroupVersionResource(client)
+		if err != nil {
+			log.Errorf("unable to discover HPA GroupVersionResource: %s", err)
+		} else {
+			genericInformerFactory, err := informer.ForResource(hpaGVR)
+			if err != nil {
+				log.Errorf("error creating generic informer: %s", err)
+			}
+
+			autoscalerLister = genericInformerFactory.Lister()
+			autoscalerListerSynced = genericInformerFactory.Informer().HasSynced
```

Comment:
Shouldn't we do this only when `informer.ForResource(hpaGVR)` returns a nil error?